spec/numberformat.html

```diff
@@ -23,26 +30,43 @@ <h1>SetNumberFormatDigitOptions ( _intlObj_, _options_, _mnfdDefault_, _mxfdDefa
         1. Let _mxsd_ be ? Get(_options_, *"maximumSignificantDigits"*).
         1. Set _intlObj_.[[MinimumIntegerDigits]] to _mnid_.
         1. If _mnsd_ is not *undefined* or _mxsd_ is not *undefined*, then
-          1. Set _intlObj_.[[RoundingType]] to ~significantDigits~.
-          1. Let _mnsd_ be ? DefaultNumberOption(_mnsd_, 1, 21, 1).
-          1. Let _mxsd_ be ? DefaultNumberOption(_mxsd_, _mnsd_, 21, 21).
+          1. Let _hasSd_ be *true*.
+        1. Else,
+          1. Let _hasSd_ be *false*.
+        1. If _mnfd_ is not *undefined* or _mxfd_ is not *undefined*, then
+          1. Let _hasFd_ be *true*.
+        1. Else,
+          1. Let _hasFd_ be *false*.
+        1. Let _needSd_ be _hasSd_.
+        1. If _hasSd_ is *false* and (_hasFd_ is *true* or _notation_ is not *"compact"*), then
+          1. Let _needFd_ be *true*.
+        1. Else,
+          1. Let _needFd_ be *false*.
+        1. If _needSd_ is *true*, then
+          1. Assert: _hasSd_ is *true*.
+          1. Set _mnsd_ to ? DefaultNumberOption(_mnsd_, 1, 21, 1).
+          1. Set _mxsd_ to ? DefaultNumberOption(_mxsd_, _mnsd_, 21, 21).
           1. Set _intlObj_.[[MinimumSignificantDigits]] to _mnsd_.
           1. Set _intlObj_.[[MaximumSignificantDigits]] to _mxsd_.
-        1. Else if _mnfd_ is not *undefined* or _mxfd_ is not *undefined*, then
-          1. Set _intlObj_.[[RoundingType]] to ~fractionDigits~.
-          1. Let _mnfd_ be ? DefaultNumberOption(_mnfd_, 0, 20, *undefined*).
-          1. Let _mxfd_ be ? DefaultNumberOption(_mxfd_, 0, 20, *undefined*).
-          1. If _mnfd_ is *undefined*, set _mnfd_ to min(_mnfdDefault_, _mxfd_).
-          1. Else if _mxfd_ is *undefined*, set _mxfd_ to max(_mxfdDefault_, _mnfd_).
-          1. Else if _mnfd_ is greater than _mxfd_, throw a *RangeError* exception.
-          1. Set _intlObj_.[[MinimumFractionDigits]] to _mnfd_.
-          1. Set _intlObj_.[[MaximumFractionDigits]] to _mxfd_.
-        1. Else if _notation_ is *"compact"*, then
-          1. Set _intlObj_.[[RoundingType]] to ~compactRounding~.
+        1. If _needFd_ is *true*, then
+          1. If _hasFd_ is *true*, then
+            1. Set _mnfd_ to ? DefaultNumberOption(_mnfd_, 0, 20, *undefined*).
+            1. Set _mxfd_ to ? DefaultNumberOption(_mxfd_, 0, 20, *undefined*).
+            1. If _mnfd_ is *undefined*, set _mnfd_ to min(_mnfdDefault_, _mxfd_).
+            1. Else if _mxfd_ is *undefined*, set _mxfd_ to max(_mxfdDefault_, _mnfd_).
+            1. Else if _mnfd_ is greater than _mxfd_, throw a *RangeError* exception.
+            1. Set _intlObj_.[[MinimumFractionDigits]] to _mnfd_.
+            1. Set _intlObj_.[[MaximumFractionDigits]] to _mxfd_.
+          1. Else,
+            1. Set _intlObj_.[[MinimumFractionDigits]] to _mnfdDefault_.
+            1. Set _intlObj_.[[MaximumFractionDigits]] to _mxfdDefault_.
+        1. If _needSd_ or _needFd_, then
+          1. If _hasSd_, then
```

Comment:
```suggestion
        1. If _needSd_ is *true* or _needFd_ is *true*, then
          1. If _hasSd_ is *true*, then
```
But for that matter, this part seems hard to follow. Broader suggestion:
```
        1. If _needSd_ is *false* and _needFd_ is *false*, then
            1. Set _intlObj_.[[RoundingType]] to ~compactRounding~.
        1. Else if _hasSd_ is *true*, then
            1. Set _intlObj_.[[RoundingType]] to ~significantDigits~.
        1. Else,
            1. Set _intlObj_.[[RoundingType]] to ~fractionDigits~.
```